lambdas/tests/unit/services/test_process_mns_message_service.py

```diff
@@ -0,0 +1,155 @@
+from unittest.mock import call
+
+import pytest
+from enums.patient_ods_inactive_status import PatientOdsInactiveStatus
+from models.mns_sqs_message import MNSSQSMessage
+from services.process_mns_message_service import MNSNotificationService
+from tests.unit.conftest import TEST_CURRENT_GP_ODS
+from tests.unit.handlers.test_mns_notification_handler import (
+    MOCK_DEATH_MESSAGE_BODY,
+    MOCK_GP_CHANGE_MESSAGE_BODY,
+    MOCK_INFORMAL_DEATH_MESSAGE_BODY,
+    MOCK_REMOVED_DEATH_MESSAGE_BODY,
+)
+from tests.unit.helpers.data.dynamo_responses import (
+    MOCK_EMPTY_RESPONSE,
+    MOCK_SEARCH_RESPONSE,
+)
+from utils.exceptions import PdsErrorException
+
+
+@pytest.fixture
+def mns_service(mocker, set_env, monkeypatch):
+    monkeypatch.setenv("PDS_FHIR_IS_STUBBED", "False")
+    service = MNSNotificationService()
+    mocker.patch.object(service, "dynamo_service")
+    mocker.patch.object(service, "get_updated_gp_ods")
+    mocker.patch.object(service, "sqs_service")
+    yield service
+
+
+gp_change_message = MNSSQSMessage(**MOCK_GP_CHANGE_MESSAGE_BODY)
+death_notification_message = MNSSQSMessage(**MOCK_DEATH_MESSAGE_BODY)
+informal_death_notification_message = MNSSQSMessage(**MOCK_INFORMAL_DEATH_MESSAGE_BODY)
+removed_death_notification_message = MNSSQSMessage(**MOCK_REMOVED_DEATH_MESSAGE_BODY)
+
+
+def test_handle_gp_change_message_called_message_type_gp_change(mns_service, mocker):
+    mocker.patch.object(mns_service, "handle_gp_change_notification")
```

Comment:
`handle_gp_change_notification` is used multiple times in this file, any way we can create a fixture and also `assert_called_with` instead of just an `assert_called()`